src/Spice86.Core/Emulator/ProgramExecutor.cs

```diff
@@ -57,14 +60,30 @@ public sealed class ProgramExecutor : IDisposable {
     /// </summary>
     public void Run() {
         _gdbServer?.StartServerAndWait();
-        Machine.Run();
         if (RecordData) {
-            new RecorderDataWriter(_configuration.RecordedDataDirectory, Machine,
-                _loggerService)
-                .DumpAll();
+            DumpEmulatorStateToDirectory(_configuration.RecordedDataDirectory);
         }
+        _emulationLoop.Run();
     }
 
+    public State CpuState => Machine.Cpu.State;
+
+    public VideoState VideoState => Machine.VgaRegisters;
+
+    public ArgbPalette ArgbPalette => Machine.VgaRegisters.DacRegisters.ArgbPalette;
```

Comment:
What is the purpose of these public properties? They do not seem to be related to the concept of a programexecutor.
Is this the ProgramExecuter taking over some of the "servicecontainer" functionality of the Machine?

Reply:
This is a temporary fix so I could remove all references that where left in the MainWindowViewModel to the Machine.